packages/web-components/lib/components/selfie/src/SelfieCaptureScreens.js

```diff
@@ -233,110 +204,139 @@ class SelfieCaptureScreens extends HTMLElement {
   }
 
   // Force remount of selfie-capture-wrapper component for clean state
-  forceWrapperRemount() {
+  async forceWrapperRemount() {
+    SmartCamera.stopMedia();
+
     this._remountKey++;
+
     const container = this.querySelector('div');
-    const oldSelfieCapture = this.selfieCapture;
-
-    // Create new selfie-capture-wrapper element with same attributes
-    const newSelfieCapture = document.createElement('selfie-capture-wrapper');
-    newSelfieCapture.setAttribute('theme-color', this.themeColor);
-    newSelfieCapture.setAttribute('key', this._remountKey.toString());
-    newSelfieCapture.setAttribute('start-countdown', 'false');
-
-    // Copy all the correct attributes based on the getter logic
-    const showNav = this.hasAttribute('show-navigation');
-    const allowAgent = this.getAttribute('allow-agent-mode') === 'true';
-    const allowAgentTests = this.hasAttribute('show-agent-mode-for-tests');
-    const hideAttr = this.hasAttribute('hide-attribution');
-    const disableTests = this.hasAttribute('disable-image-tests');
-
-    if (showNav) newSelfieCapture.setAttribute('show-navigation', '');
-    if (allowAgent) newSelfieCapture.setAttribute('allow-agent-mode', 'true');
-    if (allowAgentTests)
-      newSelfieCapture.setAttribute('show-agent-mode-for-tests', '');
-    if (hideAttr) newSelfieCapture.setAttribute('hide-attribution', '');
-    if (disableTests) newSelfieCapture.setAttribute('disable-image-tests', '');
-    newSelfieCapture.setAttribute('hidden', '');
-
-    // Replace old element with new one
-    container.replaceChild(newSelfieCapture, oldSelfieCapture);
-
-    // Update reference
-    this.selfieCapture = newSelfieCapture;
-
-    // Re-add event listeners for the new element
-    this.setupSelfieWrapperEventListeners();
+    const oldWrapper = this.selfieCapture;
+
+    if (oldWrapper && container) {
+      // recreate wrapper element
+      const newWrapper = document.createElement('selfie-capture-wrapper');
+
+      // copy attributes from old wrapper, but skip key and start-countdown
+      const attributesToCopy = Array.from(oldWrapper.attributes).filter(
+        (attr) => attr.name !== 'key' && attr.name !== 'start-countdown',
+      );
```

Comment:
was wondering why the filter since we override this on line 232
Not really an issue just clarifying

Reply:
yeah it's a bit redundant. i think the order of the code changed a bit, but it was probably needed at some point